wdl-analysis/src/handlers/goto_definition.rs

```diff
@@ -0,0 +1,611 @@
+                    (ns.source().as_ref(), lines)
+                }
+                None => (document_uri, lines.clone()),
+            };
+
+            let struct_node =
+                v1::StructDefinition::cast(SyntaxNode::new_root(struct_def.node().clone()))
+                    .expect("should cast to struct definition");
+
+            if let Some(member) = struct_node
+                .members()
+                .find(|m| m.name().text() == member_ident.text())
+            {
+                let member_span = member.name().span();
+                let span = Span::new(member_span.start() + struct_def.offset(), member_span.len());
+                return Ok(Some(location(uri, span, &def_lines)?));
+            }
+        }
+
+        if let Some(call_ty) = target_type.as_call() {
+            let Some(output) = call_ty.outputs().get(member_ident.text()) else {
+                return Ok(None);
+            };
+
+            let (uri, callee_lines) = match call_ty.namespace() {
+                Some(ns_name) => {
+                    let ns = analysis_doc.namespace(ns_name).unwrap();
+                    let imported_node = graph.get(graph.get_index(ns.source()).unwrap());
+                    let lines = imported_node.parse_state().lines().unwrap().clone();
+                    (ns.source().as_ref(), lines.clone())
+                }
+                None => (document_uri, lines.clone()),
+            };
+
+            return Ok(Some(location(uri, output.name_span(), &callee_lines)?));
+        }
+
+        Ok(None)
+    }
+}
+
+/// Finds an identifier token at the specified `TextSize` offset in the concrete
+/// syntax tree.
+fn find_identifier_token_at_offset(node: &SyntaxNode, offset: TextSize) -> Option<SyntaxToken> {
+    node.token_at_offset(offset)
+        .find(|t| t.kind() == SyntaxKind::Ident)
+}
+
+/// Converts a text size offset to LSP position.
+fn position(index: &LineIndex, offset: TextSize) -> Result<Position> {
+    let line_col = index.line_col(offset);
+    let line_col = index
+        .to_wide(WideEncoding::Utf16, line_col)
+        .with_context(|| {
+            format!(
+                "invalid line column: {line}:{column}",
+                line = line_col.line,
+                column = line_col.col
+            )
+        })?;
+
+    Ok(Position::new(line_col.line, line_col.col))
+}
+
+/// Converts a `Span` to an LSP location
+fn location(uri: &Url, span: Span, lines: &Arc<LineIndex>) -> Result<GotoDefinitionResponse> {
+    let start_offset = TextSize::from(span.start() as u32);
+    let end_offset = TextSize::from(span.end() as u32);
+    let range = lsp_types::Range {
+        start: position(lines, start_offset)?,
+        end: position(lines, end_offset)?,
+    };
+
+    Ok(GotoDefinitionResponse::Scalar(Location::new(
+        uri.clone(),
+        range,
+    )))
+}
+
+/// Finds global structs, tasks and workflow definition in a document
```

Comment:
Please make all of these end in a period. I see a few.

```suggestion
/// Finds global structs, tasks and workflow definition in a document.
```